src/sentry/static/sentry/app/components/bulkNotice.tsx

```diff
@@ -0,0 +1,123 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import Alert from 'app/components/alert';
+import {tn, tct, t} from 'app/locale';
+import {defined} from 'app/utils';
+
+export function getEverythingSelectedText(allRowsCount?: number, bulkLimit?: number) {
+  if (!defined(allRowsCount)) {
+    return t('Selected all items across all pages.');
+  }
+  if (bulkLimit && allRowsCount > bulkLimit) {
+    return tct('Selected up to the first [count] items.', {
+      count: bulkLimit,
+    });
+  }
+
+  return tct('Selected all [count] items.', {
+    count: allRowsCount,
+  });
+}
+
+export function getSelectEverythingText(allRowsCount?: number, bulkLimit?: number) {
+  if (!defined(allRowsCount)) {
+    return t('Select all items across all pages.');
+  }
+  if (bulkLimit && allRowsCount > bulkLimit) {
+    return tct('Select the first [count] items.', {
+      count: bulkLimit,
+    });
+  }
+
+  return tct('Select all [count] items.', {
+    count: allRowsCount,
+  });
+}
+
+type Props = {
+  /**
+   * Number of selected rows
+   */
+  selectedRowsCount: number;
+  /**
+   * Are all rows on current page selected?
+   */
+  isPageSelected: boolean;
+  /**
+   * Are all rows across all pages selected?
+   */
+  isEverythingSelected: boolean;
+  /**
+   * Callback to select all rows across all pages
+   */
+  onSelectAllRows: () => void;
+  /**
+   * Callback to clear selection of all rows
+   */
+  onCancelAllRows: () => void;
+  /**
+   * Number of columns to stretch the notice (used in grid-column)
+   */
+  columnsCount: number;
+  /**
+   * Number of max items we can perform bulk operations on, defaults to 1000
+   */
+  bulkLimit?: number;
+  /**
+   * Number of all rows across all pages
+   */
+  allRowsCount?: number;
+  className?: string;
+};
+
+function BulkNotice({
```

Comment:
here you return either null or a react element. Please why did you decide to use a function instead of a functional component?

Reply:
This _is_ function component, it's just not declared as a const, but function.

Reply:
yes that's what I meant ... shouldn't we use const in this case?